### ISSUE TITLE
Handle deploy for multiple images(artifacts) in helm values

### ISSUE THREAD

**Issue by AMecea** [GoogleContainerTools/skaffold#193]
Hello, I am working on a project that has multiple docker images and I pass those images to helm through chart values. For the first time the `dev` command works well, but when I modify some files that trigger the build for only one artifact the deploy fails with the following warning:
```
WARN[0069] run: deploy step: deploying test: matching build results to chart values: No build present for image1
```
But I was expecting to deploy the new values with the new image. I think that the function [JoinTagsToBuildResult](https://github.com/GoogleCloudPlatform/skaffold/blob/master/pkg/skaffold/deploy/deploy.go#L38) should match the builds with the chart values not the other way.

skaffold.yaml:
```
apiVersion: skaffold/v1
kind: Config
build:
  tagPolicy: sha256
  artifacts:
  - imageName: image1
    dockerfilePath: hack/docker/toolbox/Dockerfile
    workspace: .
  - imageName: image2
    workspace: .
  local: {}
deploy:
  helm:
    releases:
    - name: test
      chartPath: hack/charts/controller
      values:
        image: image1
        conf.toolbox-image: image2
```
skaffold version is `v0.2.0`

Thank you

**Comment by r2d4** [comment]
I'm going to call this something of a bug since its a regression since we've added conditional builds.  Basically, on the first run, everything is built but on the next run, only the changed artifacts are build. The deploy actually needs to know about the latest of _all_ built artifacts so that we can template the case of many builds to one deployment.

**Comment by AMecea** [comment]
With version `0.3.0` half of the issue is solved. But now if I have extra values for the helm chart that are not images and are not specified in artifacts: the deploy fails.

Config example:
```
apiVersion: skaffold/v1
kind: Config
build:
  tagPolicy: sha256
  artifacts:
  - imageName: image1
    dockerfilePath: Dockerfile
  local: {}
deploy:
  helm:
    releases:
    - name: test
      chartPath: hack/charts/controller
      values:
        image: image1
        something: else
```
Error message:
```
Helm release test not installed. Installing...WARN[0001] run: deploy: deploy step: deploying test: matching build results to chart values: No build present for else
```

Do I miss something? How can I include extra chart configs?

Thank you!

**Comment by booninite** [comment]
We're doing it via `setValues` for now, if you're still having the issue @AMecea

**Comment by priyawadhwa** [comment]
I'd suggest trying out a more recent version of skaffold; we've fixed a bunch of bugs since `v0.2.0` :) 

I'm going to go ahead and close this issue due to inactivity, but please comment or reopen if you continue to experience this issue!
